net/tstun/tun_linux.go

```diff
@@ -36,6 +37,14 @@ func diagnoseLinuxTUNFailure(tunName string, logf logger.Logf, createErr error)
 	kernel := utsReleaseField(&un)
 	logf("Linux kernel version: %s", kernel)
 
+	const dev = "/dev/net/tun"
+	// Check for read/write access to /dev/net/tun before the modprobe check because modprobe would almost always
+	// fail in containerized environments, see https://github.com/tailscale/tailscale/issues/14277
+	if err := unix.Access(dev, unix.O_RDWR); err != nil {
+		logf("tailscaled likely does not have read or write access to %q, access check failed with: %v", dev, err)
+		return
+	}
```

Comment:
this doesn't seem right. the point of the "modprobe tun" on line 48 is to load the module to bring `/dev/net/tun` into existence. If /dev/net/tun doesn't exist at all, the access check will fail won't it? And then line 45 returns before it ever gets around to even trying the "modprobe tun".

So I feel like this breaks the non-containerized environments too.